.evergreen/docker/Dockerfile.musl

```diff
@@ -12,10 +12,3 @@ git --version
 c++ --version
```

Comment:
Can't comment because it is out of the diff.  Can we update the "Running the tests locally" section of the readme to include instructions that use drivers-evergreen-tools, instead of ./test/tools/cluster_setup.sh?  (and while we're at it - can we remove cluster_setup.sh and any mentions of it?)